doc/index.md

```diff
@@ -1,100 +1,178 @@
+
+Cqueues can be used in conjunction with lua-http to integrate other features into your lua application and create powerful, performant, web enabled applications. Some of the examples in this guide will use cqueues for simple demonstrations. For more resources about cqueues, please see:
+
+- [The cqueues website](http://25thandclement.com/~william/projects/cqueues.html) for more information about library. Other resources:
+
+- Cqueue examples can be found with the cqueues source code available through [git or archives](http://www.25thandclement.com/~william/projects/cqueues.html#download). 
+
+- For more information on integrating cqueues with other event loop libraries please see [integration with other event loops](https://github.com/wahern/cqueues/wiki/Integrations-with-other-main-loops).
+
+- [Other libraries that use cqueues](https://github.com/wahern/cqueues/wiki/Libraries-that-use-cqueues)
+
+## Conventions
+
+HTTP 1 request and status line fields are passed around inside of [headers](#http.headers) objects under keys `":authority"`, `":method"`, `":path"`, `":scheme"` and `":status"` as defined in HTTP 2. As such, they are all kept in string form (important to remember for the `:status` field).
+
+Header fields should always be used with lower case keys.
+
+
+### Errors
+
+- Invalid function parameters will throw a lua error (if validated).
+
+- Errors are returned as `nil, error, errno` unless noted otherwise.
+
+- Some HTTP 2 operations return/throw special [http 2 error objects](#http.h2_error).
+
+
+## Terminology
+
+Much lua-http terminology is borrowed from HTTP 2.
+
+[Connection"](#connection) - An abstraction over an underlying TCP/IP socket. Lua-http has two connection types: one for HTTP 1, one for HTTP 2.
+
+[Stream"](#stream) - A request/response on a connection object. Lua-http has two stream types: one for [HTTP 1 streams](#http.h1_stream), and one for [HTTP 2 streams](#http.h2_stream). They common interfaces are descrined in [*stream*](#stream) and [*http.stream_common*](#http.stream_common).
 
 
 # Interfaces
 
+
 ## connection
 
 All connection types expose the fields:
 
+
 ### `connection.type` <!-- --> {#connection.type}
 
-Either `"client"` or `"server"`
+The mode of use for the connection object. Valid values are:
+- `"client"` - Connects to a remote uri
+- `"server"` - Listens for connection on a local uri
 
 
 ### `connection.version` <!-- --> {#connection.version}
 
-The HTTP version as a number
+Returns the HTTP version as a number. 
 
 
 ### `connection:connect(timeout)` <!-- --> {#connection:connect}
 
+Blocks (polls?) until the connection is complete or timeout is exceeded. Connecting may include DNS lookups, TLS negotiation and HTTP2 settings exchange. Returns true on success. On failure returns nil and an error message.
```

Comment:
On failure it returns `nil, error_message, errno`. However this doesn't need to be mentioned (it should be assumed)